### ISSUE TITLE
Recent change in 3.1.1 breaks have_exact_props

### ISSUE THREAD

**Issue by mrsweaters** [inertiajs/inertia-rails#103]
The [recent change](https://github.com/inertiajs/inertia-rails/commit/7fa0510f1cfc220b9af5942e8950d50c0eac091f) to fix broken partial reloads causes shared props to come in with symbols and the rest to be strings when using `have_exact_props` in a request spec.

```rb

class SomeController < ApplicationController
  inertia_share someProperty: -> {
    {
      property_a: "some value",
      property_b: "this value"
    }
  }

  def some_method
     render inertia: "Manager/AwayServices", props: { property_c: "some other value" }
  end
end
```
If I then test this using:
```rb
expect_inertia.to have_exact_props({
  someProperty: {
    property_a: "some value",
    property_b: "this value",
  },
  property_c: "some other value"
})
```

I end up with a mismatch because the shared properties have symbols and the props in the method have strings for the keys. This worked fine in 3.0.0. I haven't tested 3.1.0.

**Comment by BrandonShar** [comment]
cc @bknoles

**Comment by bknoles** [comment]
I've got a fix up for this in #105 ! Thanks for the detailed test case. That made it easy to verify the bug so I could spend my time baffled by `HashWithIndifferentAccess` instead of trying to reproduce the issue.

**Comment by bknoles** [comment]
@mrsweaters This should be fixed in `3.1.2`. Please holler here if you are still seeing issues

**Comment by mrsweaters** [comment]
@bknoles Thank you for the fix and the excellent explanation!
